inc/fields/class-field-post-select.php

```diff
@@ -38,8 +38,10 @@ public function filter_shortcode_ui_fields( $fields ) {
 
 	public function action_enqueue_shortcode_ui() {
 
-		wp_enqueue_script( 'select2' );
-		wp_enqueue_style( 'select2' );
+		$noconflict = defined( 'SELECT2_NOCONFLICT' ) && SELECT2_NOCONFLICT;
+
+		wp_enqueue_script( $noconflict ? 'select2v4' : 'select2' );
+		wp_enqueue_style( $noconflict ? 'select2v4' : 'select2' );
```

Comment:
One thing that I think might help clean up the code would be to define the Select2 handle in use as a public static variable on the `Shortcode_UI` class. This would get around the need to repeat this pattern of checking the constant in four different places.